### ISSUE TITLE
System Does not boot to the DE - XFCE, KDE

### ISSUE THREAD

**Issue by rbharathkumar** [clearlinux/distribution#376]
**Describe the bug**
Unable to boot into the XFCE, KDE environments.

**To Reproduce**
1. Use the latest installer to install the basic bundles. 
2. The system boots to a console.
3. Update the system.
4. Install the bundles - LightDM and Xfce4_Desktop in that order.
5. Run the commands
    sudo systemctl enable lightdm
    sudo systemctl start lightdm

The system does not boot into the XFCE DE, rather get an error message requesting to go through the systemctl status lightdm and journalctl entries.

The systemctl entries show errors on the DBUS setup and error message showing that lightdm has been triggered multiple times. 

The command startx however boots to the XFCE environment, although with no sound (Opened a different Issue for it).

**Expected behavior**
Expect to boot into the XFCE environment.
Installing the Xfce4_Desktop/desktop-kde should also install a pre-configured Login Manager automatically unless there is a different Login Manager installed.

**Comment by gvancuts** [comment]
Just to make sure, this is in a VirtualBox VM, isn't it? Or have you seen the same issue running on bare-metal hardware?

**Comment by rbharathkumar** [comment]
Thank you for your response. 
Yes, this is in a VirtualBox. Wanted to try XFCE/KDE before trying it out on bare metal.

**Comment by anselmolsm** [comment]
@rbharathkumar desktop-kde installs sddm as the default login manager, which received some config updates lately. Have you tried that one?

**Comment by rbharathkumar** [comment]
Hi @anselmolsm . I have not tried KDE in ClearLinux in a month or so. Back then it did have the same issue. Maybe I will try testing KDE out over the weekend. I had tried the XFCE install on Monday and had to go ahead with the default GNOME install since I could not get it to work.

**Comment by ahkok** [comment]
what are the lightdm journal messages?

**Comment by rbharathkumar** [comment]
Hi @ahkok ,
Please find the requested details below.
Sorry for the late response. Had to create a new VM and perform the full install to reproduce the error.
![journalctl_logs](https://user-images.githubusercontent.com/38585383/51654281-21ed7180-1fbd-11e9-8d36-5ce878e4a9e0.jpeg)
![lightdm_error_msg](https://user-images.githubusercontent.com/38585383/51654282-21ed7180-1fbd-11e9-9be5-a97dc0e52f6d.jpeg)

**Comment by ahkok** [comment]
Please use `journalctl -a` and copy/paste the relevant lightdm messages. The method you used above obscures many relevant messages like Xorg errors from the logs.

**Comment by rbharathkumar** [comment]
Since I am stuck at a console inside virtualbox, I am finding it difficult to get the whole thing. 
I also understand that help may not be possible without complete information.
Hence will respond with the complete set of screenshots soon. Thank you.

**Comment by rbharathkumar** [comment]
I am unable to get all the journalctl entries because of the following issues.
1. Lightdm does not startup.
2. Typing startx boots to the XFCE desktop.
3. Unable to open Firefox. Getting the error.
```
XPCOMGlueLoad error for file - /home/clearlinux/firefox/libxul.so
libdbus-glib-1.so.2 - Cannot open shared object file - No such File or Directory
Couldn't load XPCOM
```
4. Thought of installing some app via Flatpak. To my surprise flathub was not configured. Hence tried to add it.
```
flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
```
This gave the error message - 
Can't load uri - https://flathub.org/repo/flathub.flatpakrepo - TLS support is not available.

I do have the network-basic bundle, tried verifying and fixing the build, regenerate the TLS certs.

5. Then tried using AppImage. Since browser was not available, downloaded using the wget command. Gave execute permission and then tried executing. It would not start. I keep getting the dbus error.

Getting an XFCE desktop to work is becoming very cumbersome. I agree I am not knowledgeable of the way things work and do not expect the OS to be fool proof, but it can be a little easier.

I suggest having a desktop-xfce bundle similar to the desktop-kde and desktop bundles. It should have the pre-configured lightdm, xfce4-desktop, pulseaudio and the necessary packages (network and other libraries) to run comfortably without searching for glib, gstreamer and all the circus.

I am still eager to try. Please guide on the next steps.

**Comment by anselmolsm** [comment]
@rbharathkumar sddm-autostart is out, it's part of desktop-kde bundle.

**Comment by rbharathkumar** [comment]
Thank you so much @anselmolsm
I will create a new VM and test out the new build.

**Comment by znmeb** [comment]
I have this on VMs installed from the latest media (https://cdn.download.clearlinux.org/releases/29480/clear/clear-29480-live-desktop.iso.xz) in KVM virtual machines. There are a number of issues. LightDM requires a data directory and a configuration file for openers. But it still doesn't work after I supply those. I can capture logfiles if you want them.

See also https://github.com/clearlinux/distribution/issues/749 - there seem to be some missing bits in the live media and systems installed from them.

**Comment by ahkok** [comment]
Finally got to this.

`lightdm` was quite broken by itself. I've gone through it left and righ, and I think I've got all the issues solved. It should be possible now to have `lightdm` all by itself without relying on `os-utils-gui` or `desktop-assets` or other bundles to get the "basics".

It'll take 2-3 days for this change to propagate out. I'll check on it next week again.

**Comment by rbharathkumar** [comment]
Hi Team,
After seeing the post entry, I did try out the XFCE desktop on a new VM. The following are my observations.
**Steps to Reproduce:**
1. Use the latest clr-installer to install clearlinux. 3 Bundles were added during install - network-basic, sysadmin-basic, user-basic. I had created a user with administrator privileges.
2. On restart I had installed - xfce4-desktop followed by lightdm and restarted.
3. I had expected it to boot into the login page directly, but it did not do so.
4. I used the following commands to get to the login page.
```
sudo systemctl enable lightdm
sudo systemctl start lightdm
```

5. It booted into a login page with no background. When I tried to login with my user credentials, it would try to login and go back to the same login page.
6. When I changed the session from Weston to XFCE, I was able to login to the desktop.
7. There were several issues within the desktop,
7.1. Since there was no weston, it gave out several warnings and errors.
7.2. Making any change to the panel caused the panel to crash and exit. I had to start xfce-panel from the terminal. This happened even when adding items to the panel. Starting the panel from the terminal caused several dbus warnings and errors.
7.3. Also received several gsettings issue saying it was not available and stuff.
7.4. I had also tried `swupd verify --fix` (which says it iis deprecated and asked me to use diagnose and repair)

I think the issues still exist with the xfce desktop and would be nice if it got fixed.
